src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -1473,7 +1473,7 @@ private static void OnBreakEvent(Interop.Sys.CtrlCode ctrlCode)
                 // on work triggered from the signal handling thread.
                 // We use a new thread rather than queueing to the ThreadPool in order to prioritize handling
                 // in case the ThreadPool is saturated.
-                Thread handlerThread = new Thread(HandleBreakEvent) { IsBackground = true };
+                Thread handlerThread = new Thread(HandleBreakEvent) { IsBackground = true, Name = ".NET Console Break" };
```

Comment:
According to @tmds in https://github.com/dotnet/runtime/pull/43679 there's a limit to the number of characters that'll be displayed for a thread name on Linux.  Given the name he chose there, I'm guessing this exceeds it.

Reply:
Yeah looks like 15
https://github.com/dotnet/runtime/blob/69e114c1abf91241a0eeecf1ecceab4711b8aa62/src/coreclr/pal/src/thread/thread.cpp#L1678
I guess portability is the reason we don't throw.
In a glance at that code we don't support macOS yet either.

Reply:
Managed code retains the untruncated value, and it looks like, without trying it, that managed profilers and debuggers do too. So truncation is not necessarily bad.
The Linux naming only appeared in b519a7d02bedcdd8c97e75ca3b930658fc21b558 and Windows in 19f2c1c1a132a05324b12714d3c05a74f2ec6975 . Before that thread name was entirely a managed concept.

Reply:
These are the unfortunate (existing) ones, they all get cropped to ".NET ThreadPool" so they're indistinguishable in native Linux tools
```
".NET ThreadPool Gate"
".NET ThreadPool Wait"
".NET ThreadPool Worker"
```
I can make these eg ".NET TP Gate" or ".NET ThreadP Gate" but that would presumably make them look ugly in managed debuggers...

Reply:
It would almost be nice, if cropping was needed for pthread_setname_np, to remove a leading ".NET" if present first...

Reply:
For the others, I truncated where there would be ambiguity. In this case eg ".NET Console Br" is OK given you get the nice name everywhere else.

Reply:
I'm inclined to leave these names as is. In a native debugger, you're already a bit more advanced and/or it's unlikely the two housekeeping threads will be problematic (presumably). In a managed debugger, it's helpful to be able to see which are the worker threads, so you can inspect the work your app is doing on them. So it's nice that they're pretty in a managed debugger, even if the names are ambiguous in a native debugger/htop.